airflow/models/dagrun.py

```diff
@@ -41,7 +41,6 @@
 from airflow.exceptions import AirflowException, TaskNotFound
 from airflow.models.base import ID_LEN, Base
 from airflow.models.taskinstance import TaskInstance as TI
```

Comment:
```suggestion
from airflow.models.taskinstance import TaskInstance as TI
from airflow.settings import task_instance_mutation_hook
```

Reply:
The fix is to quite literally delay the import, to avoid an import cycle causing confusion.